cromshell

```diff
@@ -1471,6 +1488,99 @@ function fetch-all()
   return 0
 }
 
+# Get the cost for a workflow ID:
+function cost()
+{
+  which bq &>/dev/null   
+  local r=$?
+  [[ ${r} -ne 0 ]] && error "bq does not exist.  Must install the big query command-line client." && exit 8
+
+  [ ! -e ${BQ_COST_TABLE_FILE} ] && error "Big Query cost table file does not exist.  Must populate ${BQ_COST_TABLE_FILE} with big query cost table information." && exit 9
+
+  local cost_table=$(head -n1 ${BQ_COST_TABLE_FILE})
+  local id=$1
+
+  if [[ "$(uname)" == "Darwin" ]] ; then
+    local DATE_CMD=gdate
+  else
+    local DATE_CMD=date
+  fi  
+
+  local submissionDate=$( grep "${id}" ${CROMWELL_SUBMISSIONS_FILE} | head -n1 | awk '{print $1}' )
+  local y=$( echo ${submissionDate} | sed 's#^\([0-9][0-9][0-9][0-9]\).*#\1#' )
+  local m=$( echo ${submissionDate} | sed 's#^[0-9][0-9][0-9][0-9]\([0-9][0-9]\).*#\1#' )
+  local d=$( echo ${submissionDate} | sed 's#^[0-9][0-9][0-9][0-9][0-9][0-9]\([0-9][0-9]\).*#\1#' )
+
+  local START_DATE="${y}-${m}-${d}"
+  local END_DATE=$($DATE_CMD +%Y-%m-%d -d "${START_DATE} +4 days")
+  
+  echo "Using cost table: ${cost_table}"
+  echo ""
+
+  local tmp_cost_file=$( makeTemp )
+
+  # Get the cost from Big Query:
+  bq query --use_legacy_sql=false "SELECT sum(cost) FROM \`${cost_table}\`, UNNEST(labels) WHERE value = \"cromwell-${id}\" AND _PARTITIONDATE BETWEEN \"${START_DATE}\" AND \"${END_DATE}\";" > ${tmp_cost_file}
+
+  # Display the cost:
+  head -n4 ${tmp_cost_file} | tail -n1 | tr -d '| \t'
+
+  r=$?
+  return $r
+}
+
+# Get the cost for a workflow ID:
+function cost-detailed()
+{
+  which bq &>/dev/null   
+  local r=$?
+  [[ ${r} -ne 0 ]] && error "bq does not exist.  Must install the big query command-line client." && exit 8
+
```

Comment:
Can we deduplicate all this stuff?  It seems highly redundant.

Reply:
Sure.  Fixed!